### PR TITLE
Ruby's Enumerables

### PR DESCRIPTION
**Pull request by Wusinho**
Milestones for this project were:

-Creating well-known methods built-in in Ruby from ground zero

-Learn to debug these algorithms

-Apply rubocop linters for stylying

### CODE REVIEW

**Review by Georjane — CHANGES_REQUESTED** [review]
## Status: Changes Required 🔁 🚫 
### Hi Team @Wusinho and @realisticattorney 🙌🙌
Good job on opening a valid pull request👍👍 now, so I will go ahead and give it a review.
Wow, you have done great work 👏👏👏with the enumerable methods. Almost all methods behave like the original ones 👌except for a couple that I will point out below;

Before we dive into the individual enumerable, please make sure to fix all rubocop errors. Most of them can be fixed using `rubocop -a`.

1. #### #my_none?🚩

- [ ]  enumerables #none? when a Regex is passed as an argument returns false if any of the collection matches the Regex but your #my_none? returns true when a Regex is passed and one collection matches the Regex.

- [ ] enumerables #none? when a pattern other than Regex or a Class is given returns false only if one of the collection matches the pattern but your #my_none? returns true in this case. 
See example of my_none failed test below:

enumerr

2. #### #my_inject?🚩

- [ ] enumerables #my_inject is suppose to raise a "LocalJumpError" when no block or argument is given

- [ ] enumerables #my_inject is suppose to search for the longest word in an array of strings

- [ ] enumerables #my_inject when a block is given with and without an initial value combines all elements of enum by applying a binary operation, specified by a block

- [ ] enumerables #my_inject when a symbol is specified with an initial value combines each element of the collection by applying the symbol as a named method::range

rangeerr

3. #### #multiply_els🚩

- [ ] enumerables #multiply_els accepts an array as an argument and multiplies all the elements of the array together using #my_inject

You can go ahead and fix the above changes inorder for your project to get approved!

Feel free to reach out to me on Slack 👍👍if you have any questions, **_Witah Georjane_**.

Goodluck on your project and Happy coding!💻🤝

File: lib/enumerable.rb [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+      my_each { |item| return false if item == true }
+    elsif !param.nil? && (param.is_a? Class)
+      my_each { |item| return false if [item.class, item.class.superclass].include?(param) }
+    elsif !param.nil? && param.instance_of?(Regexp)
+      my_each { |item| return false if param.match(item) }
+    else
+      my_each { |item| return true unless item == param }
+    end
+    true
+  end
```
Comment (SpaYco) [inline lib/enumerable.rb]: - [x] when a pattern other than Regex or a Class is given it returns false only if one of the collection matches the pattern

```Ruby
  words = %w[dog door rod blade]
  words[0] = 5

  words.my_none?(5)
```

_expected false
            got true_

File: lib/enumerable.rb [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    elsif operator.nil?
+      operator = param
+      param = nil
+    else
+      operator = operator.to_sym
+      my_each { |item| param = param.nil? ? item : param.send(operator, item) }
+    end
+    param
+  end
+end
```
Comment (SpaYco) [inline lib/enumerable.rb]: - [ ] when a symbol is specified without an initial value it combines each element of the collection by applying the symbol as a named method

```Ruby
   array = Array.new(100) { rand(0...9) }
   array.my_inject(:+)
```

_expected: 406
got: nil_


- [ ] when a symbol is specified without an initial value it combines each element of the collection by applying the symbol as a named method:range

```Ruby
   range= Range.new(5, 50)
   range.my_inject(:*)
```

_expected: 1267253883404724085150525340252698701849068398706688000000000000
            got: nil_

**Review by SpaYco — CHANGES_REQUESTED** [review]
# Changes Requested 

Hello Team! 👋 
Awesome job so far!! 👍 you still need to make some changes though! 

## Required Changes ♻️ 

- [ ] check below 👇 

### please make the appropriate changes and ask for another code review ✔️ 

Happy Coding!

Best, 
Aziz Mejri 
_contact me on [twitter](https://twitter.com/iSpaYco) or slack (@SpaYco) if you need any help_

**Review by voscarmv — CHANGES_REQUESTED** [review]
# Impressive work, team! 
## However, some changes are still required... 

- You only have one remaining error in your `#my_none` method:
```
 1) enumerables #my_none? when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
     Failure/Error: expect(words.my_none?(5)).to be words.none?(5)
     
       expected true
            got false
```
Please make sure that the following example works before re-submitting:
```
words = %w[dog door rod blade]
words.my_none?(5) # This should return true.
```
### Submit these changes and request a new code review! 

### Questions or comments? Reach me on [Twitter](https://twitter.com/voscarmv) or [LinkedIn](https://linkedin.com/in/oscar-mier)
# Happy Hacking!

**Review by BrittanyBlake — APPROVED** [review]
### Status: Approved ✅ 
****
### Hi Team! 👋 
### Great job implementing the requested changes!! 🎖️ It's a 👍 from me! Go ahead and merge this pull request and keep up the good work 😄
****

 Good Luck and Happy Coding! 🍀 👩🏾‍💻

**Pull request merged**